### PR TITLE
mender: Reestablish labels on the root filesystems.

### PR DESCRIPTION
**Pull request by drewmoseley**
Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Add labels to the rootfs partitions

### CODE REVIEW

File: meta-mender-core/classes/mender-part-images.bbclass [hunk 1]
```diff
@@ -125,8 +125,8 @@ EOF
     fi
 
     cat >> "$wks" <<EOF
-part --source rawcopy --sourceparams="file=${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${ARTIFACTIMG_FSTYPE}" --ondisk "$ondisk_dev" --align $alignment_kb --fixed-size ${MENDER_CALC_ROOTFS_SIZE}k $part_type_params
-part --source rawcopy --sourceparams="file=${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${ARTIFACTIMG_FSTYPE}" --ondisk "$ondisk_dev" --align $alignment_kb --fixed-size ${MENDER_CALC_ROOTFS_SIZE}k $part_type_params
+part --source rawcopy --sourceparams="file=${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${ARTIFACTIMG_FSTYPE}" --fstype=${ARTIFACTIMG_FSTYPE} --label primary --ondisk "$ondisk_dev" --align $alignment_kb --fixed-size ${MENDER_CALC_ROOTFS_SIZE}k $part_type_params
+part --source rawcopy --sourceparams="file=${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${ARTIFACTIMG_FSTYPE}" --fstype=${ARTIFACTIMG_FSTYPE} --label secondary --ondisk "$ondisk_dev" --align $alignment_kb --fixed-size ${MENDER_CALC_ROOTFS_SIZE}k $part_type_params
```
Comment (kacf) [inline meta-mender-core/classes/mender-part-images.bbclass:129]: Looks good, but should we do the same for the data partition below?

**Review by kacf — APPROVED** [review]

File: meta-mender-core/classes/mender-part-images.bbclass (hunk 1 above)
Comment (drewmoseley) [reply meta-mender-core/classes/mender-part-images.bbclass:129]: A volume label is already applied in the IMAGE_CMD_dataimg() function so it's not needed here.

**Comment by lluiscampos** [comment]
@drewmoseley any any other reader for future reference,

This change modifies the checksum on the filesystem, making it not mach anymore the checksum from the Mender Artifact generated by yocto in the same build. It has been reverted in https://github.com/mendersoftware/meta-mender/pull/1219.

The reason for not caching this problem before is that the specific test for checksum verification was wrongly skip in modern yocto branches. See https://github.com/mendersoftware/mender-image-tests/pull/32 for more details on the test framework error.

**Comment by clementperon** [comment]
@lluiscampos I'm interested to have label for the rootfs, if the checksum match is it acceptable to restore the label for rootfs?

**Comment by kacf** [comment]
I suspect some wrong assumptions were made when this was first implemented. Labels cannot be used because they are tied to the filesystem. To be usable with Mender, they need to be tied to the partition entry. Consider this scenario:

```
 ┌──────────┬─────────┬──────────┐
 │Partition │ sda2    │ sda3     │
 │Label     │ primary │ secondary│
 └──────────┴─────────┴──────────┘
```

If it was updated, it might look like this:

```
 ┌──────────┬─────────┬──────────┐
 │Partition │ sda2    │ sda3     │
 │Label     │ primary │ primary  │
 └──────────┴─────────┴──────────┘
```

Because the label follows the filesystem, not the partition.

**Comment by drewmoseley** [comment]
@kacf does that mean this patch needs to be reverted?

**Comment by kacf** [comment]
> So using the PARTUUID instead of the UUID leaves to the same issue right?

Using PARTUUID instead of UUID solves the problem. Similarly, using `--partlabel` instead of `--label` would solve the problem for labels, but I haven't checked if this actually exists.

**Pull request merged**